storage/engine/mvcc.go

```diff
@@ -985,70 +1025,57 @@ func MVCCDeleteRange(engine Engine, ms *MVCCStats, key, endKey roachpb.Key, max
 	return num, nil
 }
 
-func getScanMeta(iter Iterator, encEndKey MVCCKey, meta *MVCCMetadata) (roachpb.Key, MVCCKey, error) {
+func getScanMeta(iter Iterator, encEndKey MVCCKey, meta *MVCCMetadata) (MVCCKey, error) {
 	metaKey := iter.Key()
-	if bytes.Compare(metaKey, encEndKey) >= 0 {
-		return nil, nil, iter.Error()
-	}
-	key, ts, isValue, err := MVCCDecodeKey(metaKey)
-	if err != nil {
-		return nil, nil, err
+	if !metaKey.Less(encEndKey) {
+		return NilKey, iter.Error()
 	}
-	if isValue {
+	if metaKey.IsValue() {
 		meta.Reset()
-		meta.Timestamp = ts
-		meta.KeyBytes = int64(len(iter.unsafeKey()))
+		meta.Timestamp = metaKey.Timestamp
+		meta.KeyBytes = mvccVersionTimestampSize
```

Comment:
Actually, the previous code was correct. `KeyBytes` refers to the size of the first versioned key which is always accounted for as `mvccVersionTimestampSize`. The size of the metadata key (even for implicit metadata) is accounted for separately. I've added an explanatory comment.